src/Spice86.Core/Emulator/ReverseEngineer/ArgumentFetcher.cs

```diff
@@ -96,6 +96,11 @@ public void Get(out uint arg1, out uint arg2, out ushort arg3) {
         arg3 = _stack.Peek16(12);
     }
 
+    public void Get(out uint arg1, out uint arg2) {
```

Comment:
Why not return an ValueTuple instead of void + out parameters ?

Especially since it only returns data.

Reply:
Because you cannot overload on return values only. The arguments are used to determine which Get() method to use.

And even if you could,
```csharp
ArgumentFetcher.StringShortUshort args = _args.Get(); 
string dosPath = args.Arg1;
short flags = args.Arg2;
ushort mode = args.Arg3;
```
and
```csharp
public record struct StringShortUshort(string Arg1, short Arg2, ushort Arg3);
public StringShortUshort Get() {
    return new StringShortUshort(
        GetStringFromDsPointer(_stack.Peek16(4)),
        (short)_stack.Peek16(6),
        _stack.Peek16(8)
    );
}
```

are not clearer than 

```csharp
_args.Get(out string dosPath, out short flags, out ushort mode);
```
and
```
public void Get(out string arg1, out short arg2, out ushort arg3) {
    ushort stringPointerOffset = _stack.Peek16(4);
    arg2 = (short)_stack.Peek16(6);
    arg3 = _stack.Peek16(8);
    arg1 = GetStringFromDsPointer(stringPointerOffset);
}
```

Reply:
Oh, I see. I didn't see the other Get methods.